### PR TITLE
auto load local .eslintrc

### PR DESCRIPTION
**Pull request by CodeFalling**
Load `.eslintrc` from project's root if existed.

### CODE REVIEW

**Comment by AgIt0** [comment]
We should also make it configurable  somehow IMO, because a lot of projects I work on/I've seen keep the `.eslintrc` in a subfolder `assets` for instance

**Comment by bleggett** [comment]
This is super-needed, but I think it belongs in the general Javascript layer.

**Comment by robbyoconnor** [comment]
When will this awesomeness be merged?!?!

**Comment by robbyoconnor** [comment]
When will this awesomeness be merged?!?! I agree that this needs to be in the javascript layer and not here.

File: layers/+frameworks/react/packages.el [hunk 1]
```diff
@@ -38,6 +38,14 @@
...
     (dolist (checker '(javascript-eslint javascript-standard))
       (flycheck-add-mode checker 'react-mode)))
+  (defun react/use-eslintrc-from-project-root ()
+    (let ((rc-path (if (projectile-project-p)
+                       (concat (projectile-project-root) ".eslintrc"))))
+      (if (file-exists-p rc-path)
+          (progn
+            (message rc-path)
+          (setq flycheck-eslintrc rc-path)))))
+
```
Comment (robbyoconnor) [inline layers/+frameworks/react/packages.el]: Move to the `javascript` layer.

File: layers/+frameworks/react/packages.el [hunk 2]
```diff
@@ -51,7 +59,7 @@
       (setq-local flycheck-javascript-eslint-executable eslint)))
 
   (add-hook 'react-mode-hook #'react/use-eslint-from-node-modules)
-
+  (add-hook 'react-mode-hook #'react/use-eslintrc-from-project-root)
```
Comment (robbyoconnor) [inline layers/+frameworks/react/packages.el]: Move to the `javascript` layer.

**Comment by bleggett** [comment]
Looked into this more, and 

- Current versions of eslint automatically discover .eslintrc files by walking up the  folder tree from the file being checked.
- Current versions of flycheck just run `eslint --print-config .` which will return an exit code of 0 if eslint finds a config somewhere in the tree using the method above.
- Because of the above, the variable `flycheck-eslintrc` has been deprecated in flycheck.

So really, we shouldn't ever need to explicitly tell eslint or flycheck where the .eslintrc is, as long as it is somewhere in a parent folder or the project root, which is where eslint expects to find them.

**Comment by TheBB** [comment]
Thank you for that @bleggett!

**Comment by robbyoconnor** [comment]
@TheBB -- I'm pretty strong that this belongs in the `javascript` layer...not here...if this PR were to be accepted.

**Comment by syl20bnr** [comment]
So this PR is not needed?

If so, please merge it and then correct it in a new commit.

**Comment by robbyoconnor** [comment]
Shouldn't this be on the javascript layer though @syl20bnr?

**Comment by CodeFalling** [comment]
As @bleggett said, this PR is not needed. Should we close this?

**Comment by robbyoconnor** [comment]
Since it's in the wrong layer...yeah.

**Comment by TheBB** [comment]
Closing then since not needed, no matter what layer it's in.

**Pull request closed without merging**